core/modules/user/user.admin.inc

```diff
@@ -468,6 +470,14 @@ function user_admin_permissions($form, $form_state, $role_name = NULL) {
       '#markup' => check_plain($role->label),
       '#tree' => TRUE,
     );
+
+    // Amend the entire list of checkboxes with AIRA labels for screen readers.
+    foreach ($all_permissions as $permission_key => $permission_item) {
+      $form['checkboxes'][$role_name][$permission_key]['#attributes']['aria-label'] = t('!role permission: !permission', array(
+        '!role' => $role->label,
```

Comment:
The NVDA screen reader reads out the column title first, then reads the ARIA label, so we end up hearing the role twice. But the Chrome reader only reads the ARIA label. So I suppose this will be necessary. FYI.

Reply:
Hm, we could set `role="presentation"` on the table or `aria-hidden` on the column headers. It would be nice to avoid duplicating the same thing being read aloud.